tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java

```diff
@@ -132,7 +134,7 @@ public void setResources(Configuration yarnConf,
     shellEnv.put("CLASSPATH", classPathEnv.toString());
   }
 
-  public synchronized List<TensorFlowContainerRequest> getContainersRequests() {
+  public List<TensorFlowContainerRequest> getContainersRequests() {
```

Comment:
why is `synchrnozied` not necessary here ?

Reply:
There's only one caller - the AM. AM.scheduleTasks() calls this, and this is only done in 1 thread in the AM.